lib/Core/Site/QueryTypes/ForwardFieldRelations.php

```diff
@@ -0,0 +1,116 @@
+use RuntimeException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * todo
+ */
+class ForwardFieldRelations extends OptionsResolverBasedQueryType implements QueryType
+{
+    /**
+     * @var \Netgen\EzPlatformSiteApi\Core\Site\Plugins\FieldType\RelationResolver\Registry
+     */
+    private $relationResolverRegistry;
+
+    /**
+     * @param \Netgen\EzPlatformSiteApi\Core\Site\Plugins\FieldType\RelationResolver\Registry $relationResolverRegistry
+     */
+    public function __construct(RelationResolverRegistry $relationResolverRegistry)
+    {
+        $this->relationResolverRegistry = $relationResolverRegistry;
+    }
+
+    public static function getName()
+    {
+        return 'SiteAPI:ForwardFieldRelations';
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @throws \Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException
+     * @throws \Symfony\Component\OptionsResolver\Exception\AccessException
+     */
+    protected function configureOptions(OptionsResolver $optionsResolver)
+    {
+        $optionsResolver->setRequired([
+            'content',
+            'field_definition_identifier',
+        ]);
+
+        $optionsResolver->setDefaults([
+            'limit' => 25,
+            'offset' => 0,
+            'content_type_identifiers' => [],
+            'sort_clauses' => [],
+        ]);
+
+        $optionsResolver->setAllowedTypes('content', Content::class);
+        $optionsResolver->setAllowedTypes('field_definition_identifier', 'string');
+        $optionsResolver->setAllowedTypes('limit', 'int');
+        $optionsResolver->setAllowedTypes('offset', 'int');
+        $optionsResolver->setAllowedTypes('content_type_identifiers', 'string[]');
+        $optionsResolver->setAllowedTypes('sort_clauses', 'eZ\Publish\API\Repository\Values\Content\Query\SortClause[]');
+    }
+
+    /**
+     * todo test empty id search solr/db
+     * @inheritdoc
+     *
+     * @throws \LogicException
+     * @throws \OutOfBoundsException
+     * @throws \InvalidArgumentException
+     * @throws \RuntimeException
+     */
+    protected function doGetQuery(array $parameters)
+    {
+        /** @var \Netgen\EzPlatformSiteApi\API\Values\Content $content */
+        $content = $parameters['content'];
+        $fieldDefinitionIdentifier = $parameters['field_definition_identifier'];
+        $contentTypeIdentifiers = $parameters['content_type_identifiers'];
+        $limit = $parameters['limit'];
+        $offset = $parameters['offset'];
+        $sortClauses = $parameters['sort_clauses'];
+
+        if (!$content->hasField($fieldDefinitionIdentifier)) {
+            throw new RuntimeException(
+                "Content does not contain a field '{$fieldDefinitionIdentifier}'"
+            );
+        }
+
+        $field = $content->getField($fieldDefinitionIdentifier);
+        assert($field instanceof Field);
```

Comment:
Behavior of `assert` is dependant on PHP configuration, so it would probably not be wise to use it like this?

Docs say:

> Assertions should not be used for normal runtime operations like input parameter checks. As a rule of thumb your code should always be able to work correctly if assertion checking is not activated.

Reply:
I was trying it out, because PHPStorm is not smart enough to figure out has/get combo, so it warns about possible null value for no reason. I’ll remove it, we’ll eventually implement Null field anyway.

Reply:
Removed in https://github.com/netgen/ezplatform-site-api/pull/70/commits/4bbffd58bbf5ac4bb000de9fdcbe51c2f3c350e2